src/structs/file_descriptor.rs

```diff
@@ -152,43 +154,95 @@ impl FileDescriptor {
             } else {
                 self.path_name.file_name().into()
             };
-            let sub_name = make_subfolder_name_from_content(&self.path_name)
-                .unwrap_or_else(|_| "xxxxxxxx".to_owned());
-            let new_path = trg_path_cloned.join(sub_name).join(trg_name_cloned);
-            if fs::copy(&self.path_name, &new_path).is_ok() {
+
+            if let Some(new_path) = self.copy_file_to_tmp_subdir(&trg_path_cloned, &trg_name_cloned)
+            {
                 return new_path;
             }
         }
 
         self.path_name.clone().into()
     }
 
-    #[cfg(test)]
-    pub fn consolidate_into(&self, tmp_path: &Option<PathBuf>) -> PathBuf {
-        if tmp_path.is_some() {
-            let trg_path_cloned = tmp_path.as_ref().unwrap().clone();
-            let trg_name_cloned = if self.needs_new_name() {
-                // unwrap is OK because needs_new_name returns false when file_name is None.
-                self.file_name.as_ref().unwrap().clone()
-            } else {
-                self.path_name.file_name().into()
-            };
-            let new_path = trg_path_cloned.join(&"xxxxxxxx").join(trg_name_cloned);
-            return new_path;
+    #[cfg(not(test))]
+    fn copy_file_to_tmp_subdir(&self, tmp_path: &PathBuf, tmp_name: &PathBuf) -> Option<PathBuf> {
```

Comment:
Couple of random thoughts, not to fix now:
 - it should probably be a `Result`
 - it should probably not be in this file at all